PMD fails with XPathException but maven plugin execution still success

detected in of PRs where there was update for PMD suppression.
if there is extra `]` that definitely an error, execution on pmd plugin still pass with SUCCESS 
```
$ git diff
diff --git a/config/pmd-main.xml b/config/pmd-main.xml
index 12ff21dcb8..3c0b84017e 100644
--- a/config/pmd-main.xml
+++ b/config/pmd-main.xml
@@ -30,7 +30,7 @@
                       java.io.StringWriter,java.io.CharArrayWriter,java.io.StringReader"/>
       <!-- we do close streams in special methods -->
       <property name="violationSuppressXPath"
-               value="//ClassDeclaration[@SimpleName='DefaultLogger']
+               value="//ClassDeclaration[@SimpleName='DefaultLogger']]
                     | //ClassDeclaration[@SimpleName='Main']
                         //MethodDeclaration[@Name='createListener']
```

results:
```
 ~/github/checkstyle/checkstyle 
$ ./mvnw clean test-compile pmd:check
....
[INFO] --- plexus-component-metadata:2.2.0:generate-metadata (default) @ checkstyle ---
[INFO] Discovered 10 component descriptor(s)
[INFO] 
[INFO] --- exec:3.6.3:java (default) @ checkstyle ---
[INFO] 
[INFO] --- exec:3.6.3:java (generate-search-index) @ checkstyle ---
[INFO] 
[INFO] --- resources:3.5.0:testResources (default-testResources) @ checkstyle ---
[INFO] Copying 3625 resources from src/test/resources to target/test-classes
[INFO] 
[INFO] --- compiler:3.15.0:testCompile (default-testCompile) @ checkstyle ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 910 source files with javac [debug release 21] to target/test-classes
[INFO] /home/roman/github/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java: Some input files use or override a deprecated API.
[INFO] /home/roman/github/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java: Recompile with -Xlint:deprecation for details.
[INFO] /home/roman/github/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/AbstractGuiTestSupport.java: Some input files use unchecked or unsafe operations.
[INFO] /home/roman/github/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/AbstractGuiTestSupport.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- jacoco:0.8.15:instrument (default-instrument) @ checkstyle ---
[INFO] 
[INFO] >>> pmd:3.28.0:check (default-cli) > :pmd @ checkstyle >>>
[INFO] 
[INFO] --- pmd:3.28.0:pmd (pmd) @ checkstyle ---
[INFO] PMD version: 7.26.0
[WARNING] Exception applying rule CloseResource on file /home/roman/github/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/XmlLoader.java, continuing with next rule
net.sourceforge.pmd.lang.rule.xpath.PmdXPathException: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
Exception Context:                                                                                                                                                                                                                                                                                                                                                                                       
        [1:Kind=Syntax error]                                                                                                                                                                                                                                                                                                                                                                            
        [2:Phase=INITIALIZATION]                                                                                                                                                                                                                                                                                                                                                                         
        [3:Expression=//ClassDeclaration[@SimpleName='DefaultLogger']]                     | //ClassDeclaration[@SimpleName='Main']                         //MethodDeclaration[@Name='createListener']                     | //ClassDeclaration[@SimpleName='PropertyCacheFile']                         //MethodDeclaration[@Name='persist' or @Name='serialize']                     | //ClassDeclaration[@SimpleName='XmlLoader']                         //MethodDeclaration[@Name='resolveEntity']                     | //ClassDeclaration[@SimpleName='CheckstyleAntTask']                         //MethodDeclaration[@Name='getListeners' or @Name='createDefaultLogger']                     | //ClassDeclaration[@SimpleName='AbstractNode']                         //MethodDeclaration[@Name='iterateAxis']]                                                                                                                                                                                                                                                                                                                                                                                                 
        [4:Version=3.1]                                                                                                                                                                                                                                                                                                                                                                                  
        [5:Location in expr=; Line#: -1; Column#: 47]                                                                                                                                                                                                                                                                                                                                                    
        [6:Rule applied on node=!debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle;                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                                                                         
import java.io.IOException;                                                                                                                                                                                                                                                                                                                                                                              
import java.io.InputStream;                                                                                                                                                                                                                                                                                                                                                                              
import java.util.Map;                                                                                                                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                                                                                                         
import javax.xml.p(truncated)]                                                                                                                                                                                                                                                                                                                                                                           
---------------------------------                                                                                                                                                                                                                                                                                                                                                                        
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.wrapException (SaxonXPathRuleQuery.java:151)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.<init> (SaxonXPathRuleQuery.java:98)
    at net.sourceforge.pmd.reporting.ViolationSuppressor$2.suppressOrNull (ViolationSuppressor.java:86)
    at net.sourceforge.pmd.reporting.ViolationSuppressor.suppressOrNull (ViolationSuppressor.java:198)
    at net.sourceforge.pmd.reporting.RuleContext.suppressOrNull (RuleContext.java:251)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:196)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:146)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithMessage (RuleContext.java:124)
    at net.sourceforge.pmd.reporting.RuleContext.addViolation (RuleContext.java:97)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.addCloseResourceViolation (CloseResourceRule.java:767)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.checkForResources (CloseResourceRule.java:222)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.visit (CloseResourceRule.java:204)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.acceptVisitor (ASTMethodDeclaration.java:57)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:9)
    at net.sourceforge.pmd.lang.java.ast.ASTClassBody.acceptVisitor (ASTClassBody.java:26)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:14)
    at net.sourceforge.pmd.lang.java.ast.ASTClassDeclaration.acceptVisitor (ASTClassDeclaration.java:39)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:8)
    at net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit.acceptVisitor (ASTCompilationUnit.java:106)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:413)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:183)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:402)
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:355)
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:342)
    at net.sf.saxon.expr.parser.XPathParser.parse (XPathParser.java:624)
    at net.sf.saxon.expr.parser.ExpressionTool.make (ExpressionTool.java:85)
    at net.sf.saxon.sxpath.XPathEvaluator.createExpression (XPathEvaluator.java:122)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.initialize (SaxonXPathRuleQuery.java:213)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.<init> (SaxonXPathRuleQuery.java:96)
    at net.sourceforge.pmd.reporting.ViolationSuppressor$2.suppressOrNull (ViolationSuppressor.java:86)
    at net.sourceforge.pmd.reporting.ViolationSuppressor.suppressOrNull (ViolationSuppressor.java:198)
    at net.sourceforge.pmd.reporting.RuleContext.suppressOrNull (RuleContext.java:251)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:196)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:146)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithMessage (RuleContext.java:124)
    at net.sourceforge.pmd.reporting.RuleContext.addViolation (RuleContext.java:97)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.addCloseResourceViolation (CloseResourceRule.java:767)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.checkForResources (CloseResourceRule.java:222)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.visit (CloseResourceRule.java:204)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.acceptVisitor (ASTMethodDeclaration.java:57)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:9)
    at net.sourceforge.pmd.lang.java.ast.ASTClassBody.acceptVisitor (ASTClassBody.java:26)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:14)
    at net.sourceforge.pmd.lang.java.ast.ASTClassDeclaration.acceptVisitor (ASTClassDeclaration.java:39)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:8)
    at net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit.acceptVisitor (ASTCompilationUnit.java:106)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:413)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:183)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
[WARNING] Exception occurred on node !debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle;

import java.io.IOException;
import java.io.InputStream;
import java.util.Map;

import javax.xml.p(truncated)
[WARNING] Exception applying rule CloseResource on file /home/roman/github/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java, continuing with next rule
net.sourceforge.pmd.lang.rule.xpath.PmdXPathException: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
Exception Context:                                                                                                                                                                                                                                                                                                                                                                                       
        [1:Kind=Syntax error]                                                                                                                                                                                                                                                                                                                                                                            
        [2:Phase=INITIALIZATION]                                                                                                                                                                                                                                                                                                                                                                         
        [3:Expression=//ClassDeclaration[@SimpleName='DefaultLogger']]                     | //ClassDeclaration[@SimpleName='Main']                         //MethodDeclaration[@Name='createListener']                     | //ClassDeclaration[@SimpleName='PropertyCacheFile']                         //MethodDeclaration[@Name='persist' or @Name='serialize']                     | //ClassDeclaration[@SimpleName='XmlLoader']                         //MethodDeclaration[@Name='resolveEntity']                     | //ClassDeclaration[@SimpleName='CheckstyleAntTask']                         //MethodDeclaration[@Name='getListeners' or @Name='createDefaultLogger']                     | //ClassDeclaration[@SimpleName='AbstractNode']                         //MethodDeclaration[@Name='iterateAxis']]                                                                                                                                                                                                                                                                                                                                                                                                 
        [4:Version=3.1]                                                                                                                                                                                                                                                                                                                                                                                  
        [5:Location in expr=; Line#: -1; Column#: 47]                                                                                                                                                                                                                                                                                                                                                    
        [6:Rule applied on node=!debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle.ant;                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                                                                         
import java.io.File;                                                                                                                                                                                                                                                                                                                                                                                     
import java.io.IOException;                                                                                                                                                                                                                                                                                                                                                                              
import java.io.InputStream;                                                                                                                                                                                                                                                                                                                                                                              
import java.io.O(truncated)]                                                                                                                                                                                                                                                                                                                                                                             
---------------------------------                                                                                                                                                                                                                                                                                                                                                                        
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.wrapException (SaxonXPathRuleQuery.java:151)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.<init> (SaxonXPathRuleQuery.java:98)
    at net.sourceforge.pmd.reporting.ViolationSuppressor$2.suppressOrNull (ViolationSuppressor.java:86)
    at net.sourceforge.pmd.reporting.ViolationSuppressor.suppressOrNull (ViolationSuppressor.java:198)
    at net.sourceforge.pmd.reporting.RuleContext.suppressOrNull (RuleContext.java:251)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:196)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:146)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithMessage (RuleContext.java:124)
    at net.sourceforge.pmd.reporting.RuleContext.addViolation (RuleContext.java:97)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.addCloseResourceViolation (CloseResourceRule.java:767)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.checkForResources (CloseResourceRule.java:222)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.visit (CloseResourceRule.java:204)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.acceptVisitor (ASTMethodDeclaration.java:57)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:9)
    at net.sourceforge.pmd.lang.java.ast.ASTClassBody.acceptVisitor (ASTClassBody.java:26)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:14)
    at net.sourceforge.pmd.lang.java.ast.ASTClassDeclaration.acceptVisitor (ASTClassDeclaration.java:39)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:8)
    at net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit.acceptVisitor (ASTCompilationUnit.java:106)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:413)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:183)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:402)
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:355)
    at net.sf.saxon.expr.parser.XPathParser.grumble (XPathParser.java:342)
    at net.sf.saxon.expr.parser.XPathParser.parse (XPathParser.java:624)
    at net.sf.saxon.expr.parser.ExpressionTool.make (ExpressionTool.java:85)
    at net.sf.saxon.sxpath.XPathEvaluator.createExpression (XPathEvaluator.java:122)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.initialize (SaxonXPathRuleQuery.java:213)
    at net.sourceforge.pmd.lang.rule.xpath.internal.SaxonXPathRuleQuery.<init> (SaxonXPathRuleQuery.java:96)
    at net.sourceforge.pmd.reporting.ViolationSuppressor$2.suppressOrNull (ViolationSuppressor.java:86)
    at net.sourceforge.pmd.reporting.ViolationSuppressor.suppressOrNull (ViolationSuppressor.java:198)
    at net.sourceforge.pmd.reporting.RuleContext.suppressOrNull (RuleContext.java:251)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:196)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithPosition (RuleContext.java:146)
    at net.sourceforge.pmd.reporting.RuleContext.addViolationWithMessage (RuleContext.java:124)
    at net.sourceforge.pmd.reporting.RuleContext.addViolation (RuleContext.java:97)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.addCloseResourceViolation (CloseResourceRule.java:767)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.checkForResources (CloseResourceRule.java:222)
    at net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule.visit (CloseResourceRule.java:204)
    at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.acceptVisitor (ASTMethodDeclaration.java:57)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:9)
    at net.sourceforge.pmd.lang.java.ast.ASTClassBody.acceptVisitor (ASTClassBody.java:26)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:14)
    at net.sourceforge.pmd.lang.java.ast.ASTClassDeclaration.acceptVisitor (ASTClassDeclaration.java:39)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.lambda$visitNode$0 (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.ast.internal.StreamImpl$2.forEach (StreamImpl.java:100)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.visitNode (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visitJavaNode (JavaVisitor.java:7)
    at net.sourceforge.pmd.lang.java.ast.JavaVisitor.visit (JavaVisitor.java:8)
    at net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit.acceptVisitor (ASTCompilationUnit.java:106)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:30)
    at net.sourceforge.pmd.lang.rule.RuleReference.apply (RuleReference.java:413)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:57)
    at net.sourceforge.pmd.lang.rule.internal.RuleSets.apply (RuleSets.java:183)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
[WARNING] Exception occurred on node !debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle.ant;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.O(truncated)
[WARNING] There are 2 PMD processing errors:
[WARNING] /home/roman/github/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/XmlLoader.java: PmdXPathException: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
Exception Context:
        [1:Kind=Syntax error]
        [2:Phase=INITIALIZATION]
        [3:Expression=//ClassDeclaration[@SimpleName='DefaultLogger']]                     | //ClassDeclaration[@SimpleName='Main']                         //MethodDeclaration[@Name='createListener']                     | //ClassDeclaration[@SimpleName='PropertyCacheFile']                         //MethodDeclaration[@Name='persist' or @Name='serialize']                     | //ClassDeclaration[@SimpleName='XmlLoader']                         //MethodDeclaration[@Name='resolveEntity']                     | //ClassDeclaration[@SimpleName='CheckstyleAntTask']                         //MethodDeclaration[@Name='getListeners' or @Name='createDefaultLogger']                     | //ClassDeclaration[@SimpleName='AbstractNode']                         //MethodDeclaration[@Name='iterateAxis']]
        [4:Version=3.1]
        [5:Location in expr=; Line#: -1; Column#: 47]
        [6:Rule applied on node=!debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle;

import java.io.IOException;
import java.io.InputStream;
import java.util.Map;

import javax.xml.p(truncated)]
---------------------------------
/home/roman/github/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java: PmdXPathException: net.sf.saxon.trans.XPathException: Unexpected token "]" beyond end of expression
Exception Context:
        [1:Kind=Syntax error]
        [2:Phase=INITIALIZATION]
        [3:Expression=//ClassDeclaration[@SimpleName='DefaultLogger']]                     | //ClassDeclaration[@SimpleName='Main']                         //MethodDeclaration[@Name='createListener']                     | //ClassDeclaration[@SimpleName='PropertyCacheFile']                         //MethodDeclaration[@Name='persist' or @Name='serialize']                     | //ClassDeclaration[@SimpleName='XmlLoader']                         //MethodDeclaration[@Name='resolveEntity']                     | //ClassDeclaration[@SimpleName='CheckstyleAntTask']                         //MethodDeclaration[@Name='getListeners' or @Name='createDefaultLogger']                     | //ClassDeclaration[@SimpleName='AbstractNode']                         //MethodDeclaration[@Name='iterateAxis']]
        [4:Version=3.1]
        [5:Location in expr=; Line#: -1; Column#: 47]
        [6:Rule applied on node=!debug only! [CompilationUnit:20:1]package com.puppycrawl.tools.checkstyle.ant;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.O(truncated)]
---------------------------------
[INFO] Rendering content with org.apache.maven.skins:maven-fluido-skin:jar:2.0.1 skin
[INFO] 
[INFO] <<< pmd:3.28.0:check (default-cli) < :pmd @ checkstyle <<<
[INFO] 
[INFO] 
[INFO] --- pmd:3.28.0:check (default-cli) @ checkstyle ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  45.187 s
[INFO] Finished at: 2026-07-21T18:11:58-07:00
[INFO] ------------------------------------------------------------------------
```

--- a/.ci/validation.sh
+++ b/.ci/validation.sh
@@ -586,22 +586,10 @@ checkstyle-and-sevntu)
   ;;
 
 spotbugs-and-pmd)
-  mkdir -p .ci-temp/spotbugs-and-pmd
-  CHECKSTYLE_DIR=$(pwd)
   export MAVEN_OPTS="-Xmx4g"
   ./mvnw -e --no-transfer-progress clean pmd:check
   ./mvnw -e --no-transfer-progress clean test-compile spotbugs:check
-  cd .ci-temp/spotbugs-and-pmd
-  grep "Processing_Errors" "$CHECKSTYLE_DIR/target/site/pmd.html" | cat > errors.log
-  RESULT=$(cat errors.log | wc -l)
-  if [[ $RESULT != 0 ]]; then
-    echo "Errors are detected in target/site/pmd.html."
-    sleep 5s
-  fi
-  cd ..
-  removeFolderWithProtectedFiles spotbugs-and-pmd
-  exit "$RESULT"
-;;
+  ;;
 
 site)
   ./mvnw -e --no-transfer-progress clean site -Pno-validations

--- a/pom.xml
+++ b/pom.xml
@@ -729,6 +729,7 @@
             <minimumTokens>20</minimumTokens>
             <skipEmptyReport>false</skipEmptyReport>
             <failOnViolation>true</failOnViolation>
+            <skipPmdError>false</skipPmdError>
             <printFailingErrors>true</printFailingErrors>
             <includeTests>true</includeTests>
             <rulesets>

